src/KafkaMessageMetadataStruct.h

```diff
@@ -0,0 +1,12 @@
+#pragma once
+
+#include <stdint.h>
+#include <string>
+
+struct KafkaMessageMetadataStruct {
```

Comment:
The order of members of a struct affects its memory efficiency, see http://katecpp.github.io/struct-members-order/
Unless changing the order affects clarity, it is best to order the fields in decreasing size. Here I would order as `int64_t`,`int64_t`,`int32_t`,`bool`,`std::string`.